src/compiler/sys.ts

```diff
@@ -796,11 +796,10 @@ namespace ts {
                         dirName,
                         (_eventName: string, relativeFileName) => {
                             // When files are deleted from disk, the triggered "rename" event would have a relativefileName of "undefined"
-                            const fileName = !isString(relativeFileName)
-                                ? undefined! // TODO: GH#18217
-                                : getNormalizedAbsolutePath(relativeFileName, dirName);
+                            if (!isString(relativeFileName)) { return; }

```

Comment:
`relativeFileName` is typed as `string`, so either this code is a noop or the type of `fsWatchDirectory` is wrong.